db/migration/1716401298509-AddTagGraph.ts

```diff
@@ -0,0 +1,55 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class AddTagGraph1716401298509 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`-- sql
+            CREATE TABLE tag_graph (
+                parentId INT NOT NULL,
+                childId INT NOT NULL,
+                weight INT NOT NULL DEFAULT 1,
```

Comment:
Should we maybe init this to 100 and sort it so that "heavier" items "sink down"? This makes it easier to modify individual elements up or down without quickly running into negative numbers which seem a bit weird. Totally minor thing of course and feel free to ignore

Reply:
I hemmed and hawed on what the metaphor for "weight" should be - density, or weight-as-in-importance.

Ultimately I went with importance because pressing the up arrow should move the object up, and things at the top should be the most important.

I'll increase the default if for no reason other than reducing the risk of shenanigans with the number 0 😁